schema/ridl/type_parser.go

```diff
@@ -0,0 +1,228 @@
+package ridl
```

Comment:
The type parser is still a bit hard to follow because of the `loop` jumps, we could get back to it later and refactor it with state functions (like the rest of the parser).